examples/wasm-audio-worklet/src/wasm_audio.rs

```diff
@@ -0,0 +1,60 @@
+use crate::dependent_module;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::JsFuture;
+use web_sys::{AudioContext, AudioWorkletNode, AudioWorkletNodeOptions};
+
+#[wasm_bindgen]
+struct WasmAudioProcessor(Box<dyn FnMut(&mut [f32]) -> bool>);
+
+#[wasm_bindgen]
+impl WasmAudioProcessor {
+    pub fn process(&mut self, buf: &mut [f32]) -> bool {
+        self.0(buf)
+    }
+}
+
+// Use wasm_audio if you have a single wasm audio processor in your application
+// whose samples should be played directly.
+pub async fn wasm_audio(
+    process: Box<dyn FnMut(&mut [f32]) -> bool>,
+) -> Result<AudioContext, JsValue> {
+    let ctx = AudioContext::new()?;
```

Comment:
If an `AudioContext` tries to play any audio before user interactions, it will have no effect in some browsers. Not sure if it's the case, just to remind about it.

Reply:
That's why ctx is passed to create_gui in web_main, so the gui can call resume on user interactions. I think this is less distracting than writing logic to create the context on first interaction. Also, this allows to load the audio worklet before the user interaction. Should I add a comment explaining this in web_main?